--- a/llvm/runtimes/CMakeLists.txt	2024-12-28 09:57:24.199761407 +0800
+++ b/llvm/runtimes/CMakeLists.txt	2024-12-28 09:55:37.513501617 +0800
@@ -2,7 +2,7 @@
 # Clang, except there already is clang/runtime directory which contains
 # similar although simpler functionality. We should figure out how to merge
 # the two files.
-
+add_custom_target("clang-resource-headers" ALL)
 set(COMMON_CMAKE_ARGS "-DHAVE_LLVM_LIT=ON;-DCLANG_RESOURCE_DIR=${CLANG_RESOURCE_DIR}")
 foreach(proj ${LLVM_ENABLE_RUNTIMES})
   set(proj_dir "${CMAKE_CURRENT_SOURCE_DIR}/../../${proj}")
