From 23ced1620221537f84b4a9b1e4c4d0f895bbf938 Mon Sep 17 00:00:00 2001
From: Kaiyang Wu <self@origincode.me>
Date: Mon, 23 Dec 2024 16:55:35 -0800
Subject: [PATCH 2/3] Disable SIGSTKFLT on mips

Signed-off-by: Kaiyang Wu <self@origincode.me>
---
 src/signal.rs | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/signal.rs b/src/signal.rs
index 9b51e6dfe..699ff455c 100644
--- a/src/signal.rs
+++ b/src/signal.rs
@@ -401,7 +401,7 @@ const SIGNAL_TABLE : &[LookupEntry] = &[
     #[cfg(any(bsd, target_os = "macos"))]
     LookupEntry::new(libc::SIGINFO,   L!("SIGINFO"), L!("Information request")),
 
-    #[cfg(target_os = "linux")]
+    #[cfg(all(target_os = "linux", not(target_arch = "mips64")))]
     LookupEntry::new(libc::SIGSTKFLT, L!("SISTKFLT"), L!("Stack fault")),
 
     #[cfg(target_os = "linux")]
-- 
2.47.1

